CIScripts/Test/TestConfigurationUtils.Tests.ps1

```diff
@@ -0,0 +1,82 @@
+. $PSScriptRoot\TestConfigurationUtils.ps1
+
+
+
+Describe "Select-ValidNetIPInterface unit tests" -Tags CI, Unit {
+        It "Both AddressFamily and SuffixOrigin match values in Select-ValidNetIPInterface" {
+            $MockedGetNetIPAddress = @{
+                AddressFamily = "IPv4";
+                SuffixOrigin = @("Dhcp", "Manual")
+            }
+
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should Be $MockedGetNetIPAddress
+        }
+
+        It "AddressFamily isn't matching" {
+            $MockedGetNetIPAddress = @{
+                AddressFamily = "IPv6";
+                SuffixOrigin = @("Dhcp", "Manual")
+            }
+
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should BeNullOrEmpty
+        }
+
+        It "SuffixOrigin isn't matching" {
+            $MockedGetNetIPAddress = @{
+                AddressFamily = "IPv4";
+                SuffixOrigin = @("WellKnown", "Link", "Random")
+            }
+
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should BeNullOrEmpty
+        }
+
+        It "Both AddressFamily and SuffixOrigin do not match" {
+            $MockedGetNetIPAddress = @{
+                AddressFamily = "IPv6";
+                SuffixOrigin = @("WellKnown", "Link", "Random")
+            }
+            
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult  | Should BeNullOrEmpty
+        }
+
+        It "Both AddressFamily and SuffixOrigin are empty strings" {
+            $MockedGetNetIPAddress = @{
+                AddressFamily = "";
+                SuffixOrigin = ""
+            }
+
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should BeNullOrEmpty
+        }
+
+        It "Pass valid/invalid object combinations into pipeline" {
```

Comment:
I wonder if we can run all these tests using this method: https://github.com/Juniper/contrail-windows-ci/pull/58#discussion_r177790308 (Parametrized test cases)

It would remove some boilerplate